pkg/disk/partition_table.go

```diff
@@ -678,6 +689,62 @@ func (pt *PartitionTable) ensureLVM() error {
 	return nil
 }
 
+// ensureBtrfs will ensure that the root partition is on a btrfs subvolume, i.e. if
+// it currently is not, it will wrap it in one
+func (pt *PartitionTable) ensureBtrfs() error {
+
+	rootPath := entityPath(pt, "/")
+	if rootPath == nil {
+		panic("no root mountpoint for PartitionTable")
+	}
+
+	// we need a /boot partition to boot btrfs, ensure one exists
+	bootPath := entityPath(pt, "/boot")
+	if bootPath == nil {
+		_, err := pt.CreateMountpoint("/boot", 512*common.MiB)
+
+		if err != nil {
+			return err
+		}
+
+		rootPath = entityPath(pt, "/")
+	}
+
+	parent := rootPath[1] // NB: entityPath has reversed order
+
+	if _, ok := parent.(*Btrfs); ok {
```

Comment:
I wonder if this could be a go type switch? i.e. 
```go
	switch part := parent.(type) {
	case *LVMLogicalVolume:
		return nil
	case *Partition:
		filesystem := part.Payload
...
	default:
		return fmt.Errorf("Unsupported parent for LVM")
	}
```